packages/fast-components-react-base/src/listbox/listbox.tsx

```diff
@@ -0,0 +1,314 @@
+
+    protected handledProps: HandledProps<ListboxHandledProps> = {
+        children: void 0,
+        managedClasses: void 0,
+        typeAheadPropName: void 0,
+    };
+
+    private rootElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    private typeAheadString: string = "";
+    private typeAheadTimer: any;
+    private typeAheadData: TypeAheadDataItem[] = [];
+
+    constructor(props: ListboxProps) {
+        super(props);
+
+        this.state = {
+            focusIndex: -1,
+        };
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                {...this.unhandledProps()}
+                ref={this.rootElement}
+                role="listbox"
+                className={this.generateClassNames()}
+                onKeyDown={this.handleMenuKeyDown}
+            >
+                {this.renderChildren()}
+            </div>
+        );
+    }
+
+    public componentDidMount(): void {
+        const children: Element[] = this.domChildren();
+        const focusIndex: number = children.findIndex(this.isFocusableElement);
+
+        if (focusIndex !== -1) {
+            this.setState({
+                focusIndex,
+            });
+        }
+    }
+
+    public componentWillUnmount(): void {
+        clearTimeout(this.typeAheadTimer);
+    }
+
+    /**
+     * Create class names
+     */
+    protected generateClassNames(): string {
+        return super.generateClassNames(get(this.props.managedClasses, "listbox"));
+    }
+
+    /**
+     * Render all child elements
+     */
+    private renderChildren(): React.ReactChild[] {
+        this.typeAheadData = [];
+        return React.Children.map(this.props.children, this.renderChild);
+    }
+
+    /**
+     * Render a single child
+     */
+    private renderChild = (
+        child: React.ReactElement<any>,
+        index: number
+    ): React.ReactChild => {
+        if (child.props[this.props.typeAheadPropName] !== undefined) {
+            this.typeAheadData.push({
+                index,
+                compareString: child.props[this.props.typeAheadPropName].toLowerCase(),
+            });
+        }
+        return React.cloneElement(child, {
+            tabIndex: index === this.state.focusIndex ? 0 : -1,
+        });
+    };
+
+    /**
+     * Determines if a given element should be focusable by the menu
+     */
+    private isFocusableElement = (element: Element): element is HTMLElement => {
+        return (
+            element instanceof HTMLElement && element.getAttribute("role") === "option"
+        );
+    };
+
+    /**
+     * Determines if a given element is disabled
+     */
+    private isDisabledElement = (element: Element): element is HTMLElement => {
+        return (
+            element instanceof HTMLElement &&
+            element.getAttribute("aria-disabled") === "true"
+        );
+    };
```

Comment:
The two methods above expect/anticipate that:
- something is being passed with a non-native role of option
- something is being passed with a non-natively supported "disabled" attribute.

Are we planning on having a list-box-item with a distinct shape that supports these internal expectations? I don't see it a part of this PR

Reply:
That listbox depends on interactable children to have a role of  "option" seems reasonable to me because the aria spec (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role) says "One or more nested options are required." under related roles.  We use roles similarly to identify valid children in context-menu.

The usage of  "aria-disabled" is exactly what we are doing in context-menu.  Although from reading here I wonder if we should be using just "disabled" instead in both places. (https://stackoverflow.com/questions/38059140/when-to-use-the-disabled-attribute-vs-the-aria-disabled-attribute-for-html-eleme)

The "select-option" component is not in this PR but it would fulfill these requirements.   The expectation would be that other components that fulfill these requirements could be valid items as well.

Reply:
@scomea I see the select-option specific thing as a likely implementation of the `listbox-item`. With this, if we publish, I only have part of what is required to create a functioning listbox. Additionally, the expectation for a user to be passing a `div` with `aria-disabled` is fine and valid per context menu; however context menu can clearly be constructed using `context-menu-items` where that assumption is true. Without a `listbox-item` I don't know that we can safely assume the above without either documenting it or providing something that meets those requirements.

I think the definition of done of this PR needs to be that a user can adequately create a functioning listbox per the spec so that this can be used in isolation outside of the context of our implementation of MSFT select.

Reply:
Listbox-item added and Listbox has been refactored to handle selection.